swift/Workflow/Sources/SubtreeManager.swift

```diff
@@ -49,8 +48,16 @@ extension WorkflowNode {
             /// as a result of this call to `render`.
             self.childWorkflows = context.usedChildWorkflows
             self.childWorkers = context.usedChildWorkers
+            self.sinks = context.sinks
 
-            /// Finally, merge all of the events together from the new set of children and start listening to them
+            /// Enable and start listening to events from sinks, child workflows, and workers.
+            for sink in self.sinks {
```

Comment:
As far as I'm aware, `Sink`s are explicitly for use in UI. As such, their events should always come in on the main thread.

We should assert using

```
dispatchPrecondition(condition: .onQueue(.main))
```

either here or somewhere in the sink plumbing.

Reply:
`Sink`s *currently* are only used for UI events. I agree with adding a precondition on these, but want to leave open option of running the entirety of a workflow on a background thread (ie: not bound to a container/UI, just from a `WorkflowHost`).

I'll go with adding it for now, and then if/when we want background support, make the appropriate changes.

Reply:
Rather than hard coding main, can you make the queue configurable, with .main as a default value?

Reply:
That's likely the long term path for this (making the queue configurable). Since we are only using this on the main queue currently, I'm going to leave it as is for the moment (and do that in a followup).

https://github.com/square/workflow/issues/364

Reply:
Added verification we are on the `DispatchQueue.workflowExectution` for all actions that come in (covering UI events, as well as asserting that's what we subscribe on for workers and subscriptions)

Reply:
We actually have the workflow queue defined in an extension somewhere, you'd want to assert that you're on that queue versus main (the current definition just resolves to main).

Reply:
Yup, that's the one I used.